qutebrowser/keyinput/modeman.py

```diff
@@ -223,6 +225,7 @@ def register(self, mode, parser):
         assert isinstance(mode, usertypes.KeyMode)
         assert parser is not None
         self._parsers[mode] = parser
+        parser.request_leave.connect(self.leave)
```

Comment:
Please add a `@pyqtSlot(usertypes.KeyMode, str)` decorator to `leave` - it works without the decorator, but in some circumstances you get funny behaviour, especially with old Qt versions (you might need to import `pyqtSlot` from `QtCore`)